src/coreclr/tools/superpmi/superpmi-shared/methodcontext.cpp

```diff
@@ -4777,7 +4777,7 @@ void MethodContext::recGetStringLiteral(CORINFO_MODULE_HANDLE module, unsigned m
     key.C = (DWORD)bufferSize;
 
     DWORD strBuf = (DWORD)-1;
-    if (buffer != nullptr)
+    if (buffer != nullptr && length != -1)
         strBuf = (DWORD)GetStringLiteral->AddBuffer((unsigned char*)buffer, (unsigned int)bufferSize);
```

Comment:
Isn't this still unconditionally recording 1 KB of data instead of only the returned string length?

Reply:
ah, right, will fix it, it also seems to be using incorrect buffer length.